lib/addressable/idna/pure.rb

```diff
@@ -146,16 +146,13 @@ def self.unicode_compose(unpacked)
       starter_cc = 256 if starter_cc != 0
       for i in 1...length
         ch = unpacked[i]
-        cc = lookup_unicode_combining_class(ch)
 
         if (starter_cc == 0 &&
             (composite = unicode_compose_pair(starter, ch)) != nil)
           starter = composite
-          startercc = lookup_unicode_combining_class(composite)
         else
           unpacked_result << starter
           starter = ch
-          startercc = cc
```

Comment:
Interesting. This was a straight port from C, so the original C probably had this unused variable too. Never noticed before. Cool.